src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getUser(todo.userId)
+      .then(setSelectedUser)
+      .finally(() => setIsLoading(false));
+  }, [todo.userId]);
```

Comment:
try this to fix ur flashing problem
```suggestion
  const [selectedUser, setSelectedUser] = useState<User | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    getUser(todo.userId)
      .then(setSelectedUser)
      .finally(() => setIsLoading(false));
  }, [todo.userId]);
```